open-craft/frontend-app-course-authoring
# feat: import tags to existing taxonomy

## Description
This PR adds a new menu item in the Taxonomy Card menu and the Taxonomy Details menu to allow users to import tags to existing Taxonomies by importing a CSV/JSON file. This overwrites the current tags.

![import-tags](https://github.com/openedx/frontend-app-course-authoring/assets/849463/5fcd42af-9293-4ff9-8928-e5cbf861280d)

## Testing instructions
1. Have an import taxonomy template ready to be imported. You could get one from below:
    - [template.csv](https://github.com/openedx/frontend-app-course-authoring/files/13322469/template.csv)
    - [template.json](https://github.com/openedx/frontend-app-course-authoring/files/13322474/template.json)
2. Go to the taxonomy list page (http://localhost:2001/taxonomies)
3. Click on the menu from an existing taxonomy card
4. You will be prompted to select a file to be imported.
5. Canceling the prompt will abort the operation
6. If the taxonomy is imported successfully, you will see an alert: Taxonomy imported successfully.
7. Importing an invalid taxonomy file, you will see an alert: Import failed - see details in the browser console. The error info will be logged in the browser console.

____
Private-ref: [FAL-3536](https://tasks.opencraft.com/browse/FAL-3536)
